### PR TITLE
[Do not merge] Task/casp 621 add default configuration for aws infrastructure nodes vpcs

### PR DESCRIPTION
**Pull request by KraProgrammer**

### CODE REVIEW

File: src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java [hunk 1]
```diff
@@ -501,8 +498,14 @@ public void handle(final Map<String, String[]> queryParams,
                             int minCount = Integer.parseInt(queryParams.get("MinCount")[0]);
                             int maxCount = Integer.parseInt(queryParams.get("MaxCount")[0]);
 
+                            String[] subnetIdArray = queryParams.get("SubnetId");
```
Comment (fhoerschlaeger) [inline src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java]: subnetIdArray  can be final

File: src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java [hunk 2]
```diff
@@ -2048,4 +2054,19 @@ private String getBlankResponseXml() {
         }
         return ret;
     }
+
+    /**
+     * Gets the VPC for a given subnetId.
```
Comment (fhoerschlaeger) [inline src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java]: VPC id

File: src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java [hunk 3]
```diff
@@ -713,16 +723,25 @@ public void Test_describeInstances() throws Exception {
...
         ec2Mocked2.setInstanceType(InstanceType.C3_8XLARGE);
+        ec2Mocked2.setSubnetId(SUBNET_ID);
 
         MockEc2Controller controller = Mockito.spy(MockEc2Controller.class);
 
         Map<String, AbstractMockEc2Instance> allMockEc2Instances = new ConcurrentHashMap<String, AbstractMockEc2Instance>();
         allMockEc2Instances.put(ec2Mocked1.getInstanceID(), ec2Mocked1);
         allMockEc2Instances.put(ec2Mocked2.getInstanceID(), ec2Mocked2);
 
+        MockSubnet subnet = new MockSubnet();
```
Comment (fhoerschlaeger) [inline src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java]: I'd extract this block to a method an call it for setup to avoid code duplication

File: src/main/java/com/tlswe/awsmock/ec2/control/MockEc2Controller.java [hunk 4]
```diff
@@ -146,12 +146,15 @@ public static MockEc2Controller getInstance() {
      *            max count of instances to run (but limited to {@link #MAX_RUN_INSTANCE_COUNT_AT_A_TIME})
      * @param maxCount
      *            min count of instances to run (should larger than 0)
+     * @param subnetId
+     *            The subnet id of new mock ec2 instance(s)
```
Comment (fhoerschlaeger) [inline src/main/java/com/tlswe/awsmock/ec2/control/MockEc2Controller.java]: This can be null. I'd add "May be null"

File: src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java [hunk 5]
```diff
@@ -510,17 +514,23 @@ public void Test_runInstances() throws Exception {
         MockEc2Controller controller = Mockito.spy(MockEc2Controller.class);
         Whitebox.setInternalState(handler, "mockEc2Controller", controller);
 
+        MockSubnet subnet = new MockSubnet();
+        subnet.setSubnetId(SUBNET_ID);
+        subnet.setVpcId(VPC_ID);
+        MockSubnetController subnetController = Mockito.spy(MockSubnetController.class);
```
Comment (fhoerschlaeger) [inline src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java]: subnetControllerMock ?

File: src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java [hunk 6]
```diff
@@ -1106,6 +1129,35 @@ public void Test_handleRunInstances() throws IOException {
         Assert.assertTrue(responseString.equals(DUMMY_XML_RESPONSE));
     }
 
+    @Test
+    public void Test_handleRunInstancesWithSubnetId() throws IOException {
```
Comment (fhoerschlaeger) [inline src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java]: Should we also add a test where subnetId is null?

Comment (KraProgrammer) [reply src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java]: This is tested with the Test_handleRunInstances test case.

File: src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java (hunk 5 above)
Comment (KraProgrammer) [reply src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java]: Fixed

File: src/main/java/com/tlswe/awsmock/ec2/control/MockEc2Controller.java (hunk 4 above)
Comment (KraProgrammer) [reply src/main/java/com/tlswe/awsmock/ec2/control/MockEc2Controller.java]: Fixed

File: src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java (hunk 3 above)
Comment (KraProgrammer) [reply src/test/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandlerTest.java]: Fixed

File: src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java (hunk 2 above)
Comment (KraProgrammer) [reply src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java]: Fixed

File: src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java (hunk 1 above)
Comment (KraProgrammer) [reply src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java]: Fixed
